hw/hw_11_task_02.py

```diff
@@ -0,0 +1,49 @@
+"""
+You are given the following code.
+
+class Order:
+    morning_discount = 0.25
+    def __init__(self, price):
+        self.price = price
+    def final_price(self):
+        return self.price - self.price * self.morning_discount
+Make it possible to use different discount programs.
+Hint: use strategy behavioural OOP pattern.
+https://refactoring.guru/design-patterns/strategy
+Example of the result call:
+def morning_discount(order):
+    ...
+def elder_discount(order):
+    ...
+order_1 = Order(100, morning_discount)
+assert order_1.final_price() == 50
+order_2 = Order(100, elder_discount)
+assert order_1.final_price() == 10
+"""
+
+from abc import ABC, abstractmethod
+
+
+class Strategy(ABC):
```

Comment:
You can use https://docs.python.org/3/library/typing.html?highlight=typing#typing.Protocol

If you change `meth` to `__call__` you can pass both classes bases and simple functions.